app/models/conversation.rb

```diff
@@ -1,5 +1,11 @@
 class Conversation < ActiveRecord::Base
-  belongs_to :topic
+  belongs_to :topic, touch: true
   belongs_to :author, class_name: 'User', foreign_key: :user_id
-  has_many :messages
+  has_many :messages, dependent: :destroy
+
+  validates_presence_of :name
```

Comment:
I'd probably add `:topic` and `:author` to this validator. I was able to create a new conversation without being logged in.